src/backend/api/handlers/insights.py

```diff
@@ -0,0 +1,54 @@
+from flask import Response
+
+from backend.api.handlers.decorators import api_authenticated
+from backend.api.handlers.helpers.profiled_jsonify import profiled_jsonify
+from backend.api.handlers.helpers.track_call import track_call_after_response
+from backend.common.consts.api_version import ApiMajorVersion
+from backend.common.decorators import cached_public
+from backend.common.models.insight import Insight
+from backend.common.queries.insight_query import (
+    InsightsByNameAndYearQuery,
+    InsightsByNameQuery,
+)
+
+
+@api_authenticated
+@cached_public
+def insights_leaderboards_all() -> Response:
```

Comment:
Is the intent here to query ALL insights, or just the overall aggregate ones (`year == 0`)?

Reply:
The intent was all, but maybe that's too much data in the future?

Reply:
What's the intended use case? Right now the individual year insights pages would presumably use `insights_leaderboards_year`, and the overall one would just use the aggregate. When do we need to fetch all of them?

Reply:
I dunno. Good point.